docs/enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,144 @@
+
+### User Stories
+
+N/A - TNA clusters a future feature in openshift, the
+[openshift enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/arbiter-clusters.md#user-stories)
+has user stories that are not specific for assisted-service.
+
+### Implementation Details/Notes/Constraints
+
+The enums for cluster's hyperthreading and disk encryption will be:
+- 'none'
+- 'masters'
+- 'arbiters'
+- 'workers'
+- 'masters,arbiters'
+- 'masters,workers'
+- 'arbiters,workers'
+- 'masters,arbiters,workers'
+- 'all' which will be the same as 'masters,arbiters,workers'
+
+When generating the ignitions for a TNA cluster, a new base ignition named
+arbiter.ign will be created. If it exists we need to patch it and upload it
+to s3. The list of patches arbiter.ign needs:
+- Set CA cert if it was provided.
+- If all the hosts in the cluster have IPv6 address then we need to add a
+  NetworkManager configuration ensuring that IPv6 DHCP requests use consistent
+  client identification.
+
+Our validations will need to be updated as follows:
+- A host can only be assigned the arbiter role if the cluster is TNA.
+- A cluster's arbiter_count must be zero or higher.
+- A cluster's number of arbiter nodes must be exactly the cluster's
+  arbiter_count, in order for the cluster to be ready for installation.
+- TNA clusters' can set control_plane_count to be 2.
+- TNA clusters' platform must be baremetal.
+- TNA clusters are expected to be TP stating from OCP 4.19.
+
+In the cluster's transition handler, the function enoughMastersAndWorkers
+needs to be updated to also check for arbiters. This is needed to correctly
+decide if the cluster is in installing/finalizing.
+
+We need to add arbiter's hardware requirements in:
+- The HW_VALIDATOR_REQUIREMENTS environment variable (by default we'll take
+  the worker's requirements to avoid issues mid-development).
+- The HostTypeHardwareRequirementsWrapper structs used by the operators.
+
+We will auto-assign the arbiter role to a host if all the following are true:
+- The host's cluster has less arbiter nodes than the cluster's arbiter_count. 
+- The host has the minimum hardware requirements for arbiter nodes.
+- Either the host doesn't have the minimum hardware requirements for master
+  nodes, or the host's cluster has enough master nodes.
+
+Outside of assisted-service, we also need to do the following:
+- After we update the API in assisted-services, we need to update the vendor
+  directory in assisted-installer-agent.
+- We need to update the function waitForMinMasterNodes in assisted-installer
+  to set the min master nodes to be the minimum of 2 (the current value)
+  and number of control planes (read it from the install-config) minus 1.
+
+Until we have this feature fully implemented, we should have an environment
+variable to decide if we allow TNA clusters (default value is false).
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+### Open Questions
+
+N/A
```

Comment:
We should add ABI concerns here. We should either ensure it works or disable it explicitly.

Reply:
Same with kube-api for the first pass. We should disable this explicitly in the controllers if we're not going to support it on the first pass.